var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -239,19 +246,20 @@ class Lua(LuaImplPackage):
         when="+pcfile",
     )
 
-    def install(self, spec, prefix):
+    def build(self, spec, prefix):
         if spec.satisfies("platform=darwin"):
             target = "macosx"
         else:
             target = "linux"
         make(
-            "INSTALL_TOP=%s" % prefix,
             "MYLDFLAGS="
             + " ".join((spec["readline"].libs.search_flags, spec["ncurses"].libs.search_flags)),
             "MYLIBS=%s" % spec["ncurses"].libs.link_flags,
-            "CC=%s -std=gnu99 %s" % (spack_cc, self.compiler.cc_pic_flag),
+            "CC={0} {1} {2}".format(spack_cc, self.compiler.c99_flag, self.compiler.cc_pic_flag),
             target,
         )
+
+    def install(self, spec, prefix):
```

Comment:
I *can* build `5.2.0` if I "undo" these changes.

Reply:
I was on `quartz` BTW.

Reply:
I personally don't care about this change. I was trying to be less hard-coded but there is no gnu99 flag function on though the xl compiler does toggle between `-std=gnu99` and `-qlanglvl=extc99` in the `c99_flag` property.

Would you prefer I toggle between the hard-coded vs this property based on the Lua version? I am willing to test until it fails and figure out the cut-off. Or move back to a hard-coded `-std=gnu99`?

Reply:
I'm not sure what c99_flag is normally populated with, but lua does require some gnu extensions so going to c99 is likely to cause problems.  If we had an option for enabling the extended mode that's what we would want.  If gnu99 works and we don't have a fail case for it, I'd leave it hardcoded since that's a surprisingly portable option (except windows), but it would be nice to clean up if the support stuff were there.

Reply:
Alternatively, I could add a `gnu99_flag` to the compilers make sense.

Reply:
Another idea... leave hard-coded `-std=gnu99` for previous versions and i could see if `c11_flag` would work on the newer ones.

Reply:
If that works it would be nice, I have a feeling they'll want gnu11 though.

Reply:
I decided this would be a scope creep of this PR and is going to be a rather large PR/effort alone.  I reverted back to the original hardcoded standard flag and will create an issue.